internal/engine/upper.go

```diff
@@ -131,40 +133,48 @@ func (u Upper) CreateManifests(ctx context.Context, manifests []model.Manifest,
 			case <-ctx.Done():
 				return ctx.Err()
 			case event := <-sw.events:
-				oldState := buildStates[event.manifest.Name]
-				buildState := oldState.NewStateWithFilesChanged(event.files)
-				buildStates[event.manifest.Name] = buildState
-
-				spurious, err := buildState.OnlySpuriousChanges()
-				if err != nil {
-					logger.Get(ctx).Infof("build watch error: %v", err)
-				}
+				if eventContainsConfigFiles(event) {
+					t, err := tiltfile.Load(tiltfile.FileName, os.Stdout)
+					if err != nil {
+						// TODO(dmiller) should we fail here, or is this OK?
+						return err
+					}
+					newManifests, err := t.GetManifestConfigs(string(event.manifest.Name))
+					if err != nil {
+						return err
+					}
+					if len(newManifests) != 1 {
+						return fmt.Errorf("Expected there to be 1 manifest for %s, got %d", event.manifest.Name, len(manifests))
+					}
+					newManifest := newManifests[0]
+					buildState := BuildStateClean
+					err = u.buildManifestFromBuildState(ctx, newManifest, buildState, buildStates)
+					if err != nil {
+						return err
+					}
+				} else {
+					oldState := buildStates[event.manifest.Name]
+					buildState := oldState.NewStateWithFilesChanged(event.files)
+					buildStates[event.manifest.Name] = buildState
 
-				if spurious {
-					// TODO(nick): I think we probably want to log when this happens?
-					continue
-				}
+					spurious, err := buildState.OnlySpuriousChanges()
+					if err != nil {
+						logger.Get(ctx).Infof("build watch error: %v", err)
+					}
 
-				u.logBuildEvent(ctx, event.manifest, buildState)
+					if spurious {
+						// TODO(nick): I think we probably want to log when this happens?
+						continue
+					}
 
-				result, err := u.b.BuildAndDeploy(
-					ctx,
-					event.manifest,
-					buildState)
-				if err != nil {
-					if isPermanentError(err) {
+					err = u.buildManifestFromBuildState(ctx, event.manifest, buildState, buildStates)
```

Comment:
i think this manifest will be stale, right?

Reply:
what about the next loop iteration?

Reply:
Aha, yeah it could be the case that the events have an old manifest.